### DISCUSSION TITLE
Decode mockoon requests

### DISCUSSION THREAD

**Discussion by KalaiJay** [mockoon/mockoon#1246] [Help]
Hi,
Request send to mockoon is an encoded string, is there a way to decode and set rules to the decoded string in mockoon. Any help or suggestion will be helpful.

**Comment by 255kb** [comment]
Hi, could you add an example of what you are sending (and how you want to use it in Mockoon)? Thanks

**Reply by KalaiJay** [reply]
Examples of my request

Plain string : D0B11X01XXXXXXXXXX    20080101FNXXXLNYYY
Encoded string: RDBCMTFBMDEwMDAwMDAwMDAwICAgICAyMDA4MDEwMR0eHEZOWFhYHExOWVlZ

My Request will send an Encoded string to mockoon.
I need to decode the encoded string using base64 decoder inside mockoon and apply rules based on the FN value which is the FirstName

Hope this is helpful, let me know if you need any additional information from my side.

Thanks in advance.

**Reply by 255kb** [reply]
Rules let you compare the query params/headers/body/etc values against something (the value field), but you cannot do any treatment on the entering value before the comparison. 

What you could do would be to do this in the template and use some helpers: 
```handlebars
{{setVar 'fn' (split (base64Decode (queryParamRaw 'p')) 'FN')}}

{{lookup @fn 1}}
```

After that you can be creative and use helpers like `includes`, `#if`, etc.

**Reply by KalaiJay** [reply]
Thanks for the response.
